components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java

```diff
@@ -148,7 +170,7 @@ public OAuth2ClientValidationResponseDTO validateClientInfo(String clientId, Str
             OAuth2Util.setClientTenatId(IdentityTenantUtil.getTenantId(appDO.getUser().getTenantDomain()));
 
             // Valid Client, No callback has provided. Use the callback provided during the registration.
-            if (callbackURI == null) {
+            if (StringUtils.isBlank(callbackURI)) {
```

Comment:
1. Not sending the 'redirect_uri' query param is allowed in the spec. 
2. If the redirect_uri query param is sent with an empty value we should consider it as an invalid request.

With this change, both cases will be considered valid. This is not the expected behaviour isn't it?

Reply:
Fixed